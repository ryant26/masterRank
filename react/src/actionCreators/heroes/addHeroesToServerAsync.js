export const addHeroesToServerAsync = (heroNames, socket) => {

    return () => {
        heroNames.forEach((heroName, i) => {
            socket.addHero(heroName, (i+1));
        });
    };
};