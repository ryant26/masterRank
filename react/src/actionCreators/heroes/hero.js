import * as HeroActionTypes from 'actiontypes/hero';

export const addHero = (hero) => {
  return {
    type: HeroActionTypes.ADD_HERO,
    hero
  };
};

export const removeHero = (hero) => {
    return {
        type: HeroActionTypes.REMOVE_HERO,
        hero
    };
};

export const addHeroes = (heroes) => {
  return {
    type: HeroActionTypes.ADD_HEROES,
      heroes
  };
};

export const clearAllHeroes = () => {
  return {
    type: HeroActionTypes.CLEAR_ALL_HEROES
  };
};