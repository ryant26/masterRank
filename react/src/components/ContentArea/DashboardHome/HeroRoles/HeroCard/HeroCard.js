import React, {
  Component
} from 'react';

import HeroStatsList from '../HeroStatsList/HeroStatsList';
import HeroImage from '../../../../HeroImage/HeroImage';
import PropTypes from 'prop-types';
import Modal from '../../../../Modal/Modal';
import Model from '../../../../../model/model';

const classNames = require('classnames');


export default class HeroCard extends Component {

    constructor(props) {
        super(props);
        this.state = {
            isStatsToggleOn: false,
            modalOpen: false,
            invitable: props.hero.platformDisplayName !== props.user.platformDisplayName
        };

        this.toggleStats = this.toggleStats.bind(this);
        this.openModal = this.openModal.bind(this);
        this.closeModal = this.closeModal.bind(this);
        this.invitePlayer = this.invitePlayer.bind(this);
    }

    toggleStats() {
        this.setState(prevState => ({
            isStatsToggleOn: !prevState.isStatsToggleOn
        }));
    }

    invitePlayer() {
        const userObject = {
            "platformDisplayName": this.props.hero.platformDisplayName,
            "heroName": this.props.hero.heroName
        };
        Model.inviteUserToGroup(userObject);
    }

    openModal() {
        this.setState(() => {
            return {
                modalOpen: true
            };
        });
    }

    closeModal() {
        this.setState(() => {
            return {
                modalOpen: false
            };
        });
    }

    render() {
        const classses = classNames({
            overlay: true,
            invitable: this.state.invitable,
        });

        let statLine;

        if(this.props.hero.stats) {
            let wins = this.props.hero.stats.wins;
            let losses = this.props.hero.stats.losses;
            let winPercentage = parseFloat(wins/(wins+losses) * 100.0).toFixed(1);
            statLine = (
                <div className="sub-title">
                    <span>{this.props.hero.skillRating} SR</span>
                    <span> | </span><span>{winPercentage}% WR</span>
                </div>
            );
        }


    return (
        <div className="HeroCard flex align-center">
            <div className="invitePlayerButton">
                <HeroImage heroName={this.props.hero.heroName}/>
                <div className={classses}>
                    <div className="plus-container flex align-center justify-center" onClick={this.invitePlayer}>
                        <i className="fa fa-plus"/>
                    </div>
                </div>
            </div>
            <div className="content flex flex-column">
                <div className="display-name">{this.props.hero.platformDisplayName}</div>
                {statLine}
            </div>
            <div className="button-primary" onClick={this.openModal}>
                <div className="button-content">
                    stats
                </div>
            </div>
            <Modal modalOpen={this.state.modalOpen} closeModal={this.closeModal}>
                <HeroStatsList hero={this.props.hero} invitable={this.state.invitable}/>
            </Modal>

        </div>
    );
    }
}

HeroCard.propTypes = {
    hero: PropTypes.shape({
        heroName: PropTypes.string.isRequired,
        platformDisplayName: PropTypes.string.isRequired,
        skillRating: PropTypes.number.isRequired,
        stats: PropTypes.shape({
            wins: PropTypes.number.isRequired,
            losses: PropTypes.number.isRequired,
        })
    }),
    user: PropTypes.shape({
        platformDisplayName: PropTypes.string.isRequired
    })
};