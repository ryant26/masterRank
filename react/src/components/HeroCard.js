import React, {
  Component
} from 'react';

import HeroCardStats from './HeroCardStats.js'
import HeroImage from './HeroImage.js'
import InvitePlayerButton from './InvitePlayerButton.js'

export default class HeroCard extends Component {

  constructor(props) {
    super(props);
    this.state = {
      isStatsToggleOn: false,
      isHovering: false,
    };

    this.toggleStats = this.toggleStats.bind(this);
    this.toggleInvitePlayerButton = this.toggleInvitePlayerButton.bind(this);
  }

  toggleStats() {
    this.setState(prevState => ({
      isStatsToggleOn: !prevState.isStatsToggleOn
    }))
  }

  toggleInvitePlayerButton() {
    this.setState(prevState => ({
      isHovering: !prevState.isHovering
    }))
  }

  render() {
    const containerStyle = {
      'display': 'flex',
      'backgroundColor': '#66ccff',
      'margin': '10px',
    }

    const statsStyle = {
      'display': 'flex',
    }

    const componentStyle = {
      'textAlign': 'center',
      'height': '60px',
      'width': '33%',
    }

    let winPercentage = parseFloat(this.props.hero.general_stats.win_percentage * 100.0).toFixed(1);

    return (
      <div>
        <div style={containerStyle}
          onMouseEnter={this.toggleInvitePlayerButton}
          onMouseLeave={this.toggleInvitePlayerButton}>

          {
            this.state.isHovering
            ? <InvitePlayerButton />
            : <HeroImage heroName={this.props.hero.name}/>
          }

          <div> Username </div>

          <div style={statsStyle} onClick={this.toggleStats}>
            <div style={componentStyle} >
              <div>Rank</div>
              <div>2734</div>
            </div>

            <div style={componentStyle}>
              <div>Games Played</div>
              <div>{this.props.hero.general_stats.games_played}</div>
            </div>

            <div style={componentStyle}>
              <div>WIN %</div>
              <div>{winPercentage}%</div>
            </div>
          </div>
        </div>
        
        {
          this.state.isStatsToggleOn
          ? <HeroCardStats hero={this.props.hero}/>
          : null
        }
      </div>
    );
  }
}
