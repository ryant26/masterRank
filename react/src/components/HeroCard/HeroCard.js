import React, {
  Component
} from 'react';

import HeroCardStats from '../HeroCardStats/HeroCardStats';
import HeroImage from '../HeroImage/HeroImage';
import InvitePlayerButton from '../InvitePlayerButton/InvitePlayerButton';
import PropTypes from 'prop-types';

export default class HeroCard extends Component {

  constructor(props) {
    super(props);
    this.state = {
      isStatsToggleOn: false,
      isHovering: false,
    };

    this.toggleStats = this.toggleStats.bind(this);
    this.toggleInvitePlayerButton = this.toggleInvitePlayerButton.bind(this);
  }

  toggleStats() {
    this.setState(prevState => ({
      isStatsToggleOn: !prevState.isStatsToggleOn
    }));
  }

  toggleInvitePlayerButton() {
    this.setState(prevState => ({
      isHovering: !prevState.isHovering
    }));
  }

  render() {
    const backgroundStyle = {
      'backgroundColor': '#66ccff'      
    };

    const containerStyle = {
      'display': 'flex',
    };

    const statsStyle = {
      'display': 'flex',
      'flexDirection':'column',
      'width':'240px',
      'cursor':'s-resize'      
    };

    const componentStyle = {
      'height':'20px'
    };
    
    let wins = this.props.hero.wins;
    let losses = this.props.hero.losses;
    let winPercentage = parseFloat(wins/(wins+losses) * 100.0).toFixed(1);

    return (
      <div>
        <div style={backgroundStyle}>
          <div onMouseEnter={this.toggleInvitePlayerButton}
               onMouseLeave={this.toggleInvitePlayerButton}>
                {
                    this.state.isHovering? <InvitePlayerButton /> : 
                                           <HeroImage heroName={this.props.hero.heroName}/>
                }
          </div>
          <div style={containerStyle}>
            <div style={statsStyle} onClick={this.toggleStats}>
              <div>
                <div><b>{this.props.hero.platformDisplayName}</b></div>
              </div>
              <div style={componentStyle} >
                <div><b>Rank:</b> 2733</div>
              </div>

              <div style={componentStyle}>
                <div><b>Hours Played:</b> {this.props.hero.hoursPlayed}</div>
              </div>

              <div style={componentStyle}>
                <div><b>Win Rate:</b> {winPercentage}%</div>
              </div>
            </div>
          </div>
        </div>
        {
          this.state.isStatsToggleOn
          ? <HeroCardStats hero={this.props.hero}/>
          : null
        }
      </div>
    );
  }
}

HeroCard.propTypes = {
  hero: PropTypes.shape({
    heroName: PropTypes.string.isRequired,
    wins: PropTypes.number.isRequired,
    losses: PropTypes.number.isRequired,
    platformDisplayName: PropTypes.string.isRequired,
    hoursPlayed: PropTypes.number.isRequired,
  })
};