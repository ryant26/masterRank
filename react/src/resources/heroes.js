const HEROES = [
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "soldier76",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "hanzo",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "dva",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
    {
        "platformDisplayName": "PwNShoPP#1662",
        "heroName": "zenyatta",
        "skillRating": 2400,
        "stats": {
            "hoursPlayed": 3,
            "wins": 11,
            "losses": 7,
            "gamesPlayed": 18,
            "kdRatio": 2.9936305732484074,
            "accuracy": 36,
            "blockedPerMin": 0,
            "healingPerMin": 148.0888888888889,
            "damagePerMin": 1338.638888888889,
            "avgObjElims": 10.5,
            "avgObjTime": 56.22222222222222,
            "pKdRatio": 1,
            "pAccuracy": 1,
            "pBlockedPerMin": 0,
            "pHealingPerMin": 1,
            "pDamagePerMin": 1,
            "pAvgObjElims": 1,
            "pAvgObjTime": 1
        }
    },
];

export default HEROES;
