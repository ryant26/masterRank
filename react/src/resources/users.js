export const users = [
    {
        "_id": 12,
        "platformDisplayName": "PwNShoPP#1662",
        "lastUpdated": "2017-10-15T01:43:36.459Z",
        "platform": "pc",
        "level": 155,
        "portrait": "https://d1u1mce87gyfbn.cloudfront.net/game/unlocks/0x025000000000115A.png",
        "region": "us"
    }
];