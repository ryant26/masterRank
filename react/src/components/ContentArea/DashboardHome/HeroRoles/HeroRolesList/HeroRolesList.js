import React, {
    Component
} from 'react';

import HeroCard from '../HeroCard/HeroCard';
import PropTypes from 'prop-types';

export default class HeroRoles extends Component {

    render() {

        const heroes = this.props.heroes;
        const role = this.props.role;

        let heroCardComponents = heroes.map((hero, i) => {
            return <HeroCard hero={hero} key={i}/>;
        });

        if (!heroCardComponents.length) {
            heroCardComponents = (
                <div className="sub-title empty-list-message flex justify-center">
                    <div>No players online</div>
                </div>
            );
        }

        return (
            <div className="HeroRolesList card flex flex-column grow">
                <div className="header flex justify-between">
                    <h3>{role}</h3>
                    <img className="role-icon"
                         src={require(`../../../../../assets/${this.props.role}.png`)}
                         alt = {this.props.role+' icon'}
                    />
                </div>
                <div className="body flex flex-column">
                    { heroCardComponents }
                </div>
            </div>
        );
    }
}

HeroRoles.propTypes = {
    heroes: PropTypes.array.isRequired,
    role: PropTypes.string.isRequired,
};