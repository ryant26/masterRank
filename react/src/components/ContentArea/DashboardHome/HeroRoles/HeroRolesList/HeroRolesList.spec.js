import React from 'react';
import { shallow } from 'enzyme';

import HeroRolesList from './HeroRolesList';
import HeroCard from '../HeroCard/HeroCard';
import heroes from '../../../../../resources/heroes';
import {users} from '../../../../../resources/users';


describe('HeroRolesList Component', () => {
    let wrapper;

    const getHeroRoles = ({heroes, role}) => {
        return shallow(
            <HeroRolesList heroes={heroes} role={role}/>
        );
    };

    beforeEach(() => {
        wrapper = getHeroRoles({
            heroes: [heroes[0]],
            role: 'tank'
        });
    });

    it('should render without exploding', () => {
        expect(wrapper).toHaveLength(1);
        expect(wrapper.find(HeroCard)).toHaveLength(1);
    });

    it('should render multiple heroCards', () => {
        wrapper = getHeroRoles({
            heroes: [heroes[0], heroes[1]],
            role: 'tank'
        });

        expect(wrapper.find(HeroCard)).toHaveLength(2);
    });

    it('should render a message when there are no hero cards', () => {
        wrapper = getHeroRoles({
            heroes: [],
            role: 'tank'
        });

        expect(wrapper.find('.empty-list-message > div').text()).toBe('No players online');
    });

    it('should render the container with hero cards sorted by winRate', () => {
        const heroWithLowerWinRate = heroes[0];
        const heroWithHigherWinRate = heroes[4];

        const lowerWinRate = heroWithLowerWinRate.stats.wins / heroWithLowerWinRate.stats.gamesPlayed;
        const higherWinRate = heroWithHigherWinRate.stats.wins / heroWithHigherWinRate.stats.gamesPlayed;

        wrapper = getHeroRoles({
            heroes: [heroWithLowerWinRate, heroWithHigherWinRate],
            role: 'tank',
            user: users[0]
        });

        expect(wrapper.find(HeroCard)).toHaveLength(2);
        expect(lowerWinRate).toBeLessThan(higherWinRate);
        expect(wrapper.find(HeroCard).first().props().hero.heroName).toBe(heroWithHigherWinRate.heroName);
    });

    describe('title should match the role prop', () => {
        const testMatchingTitleAndRole = (role) => {
            wrapper = getHeroRoles({
                heroes: [],
                role
            });

            expect(wrapper.find('h3').text()).toBe(role);
        };

        it('tank', () => {
            testMatchingTitleAndRole('tank');
        });

        it('support', () => {
            testMatchingTitleAndRole('support');
        });

        it('attack', () => {
            testMatchingTitleAndRole('attack');
        });

        it('defense', () => {
            testMatchingTitleAndRole('defense');
        });
    });
});