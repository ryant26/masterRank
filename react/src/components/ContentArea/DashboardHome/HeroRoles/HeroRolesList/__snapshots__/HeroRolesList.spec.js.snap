// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroRoles Component should match the snapshot multiple heroes 1`] = `
<div
  className="HeroRolesList card flex flex-column grow"
>
  <div
    className="header flex justify-between"
  >
    <h3>
      tank
    </h3>
    <img
      alt="tank icon"
      className="role-icon"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    className="body flex flex-column"
  >
    <div
      className="HeroCard flex align-center"
    >
      <div
        className="invitePlayerButton"
      >
        <div
          className=""
        >
          <img
            alt="soldier76 icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
        <div
          className="overlay"
        >
          <div
            className="plus-container flex align-center justify-center"
            onClick={[Function]}
          >
            <i
              className="fa fa-plus"
            />
          </div>
        </div>
      </div>
      <div
        className="content flex flex-column"
      >
        <div
          className="display-name"
        >
          You
        </div>
        <div
          className="sub-title"
        >
          <span>
            2400
             SR
          </span>
          <span>
             | 
          </span>
          <span>
            61.1
            % WR
          </span>
        </div>
      </div>
      <div
        className="button-primary"
        onClick={[Function]}
      >
        <div
          className="button-content"
        >
          stats
        </div>
      </div>
    </div>
    <div
      className="HeroCard flex align-center"
    >
      <div
        className="invitePlayerButton"
      >
        <div
          className=""
        >
          <img
            alt="hanzo icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
        <div
          className="overlay"
        >
          <div
            className="plus-container flex align-center justify-center"
            onClick={[Function]}
          >
            <i
              className="fa fa-plus"
            />
          </div>
        </div>
      </div>
      <div
        className="content flex flex-column"
      >
        <div
          className="display-name"
        >
          You
        </div>
        <div
          className="sub-title"
        >
          <span>
            2400
             SR
          </span>
          <span>
             | 
          </span>
          <span>
            61.1
            % WR
          </span>
        </div>
      </div>
      <div
        className="button-primary"
        onClick={[Function]}
      >
        <div
          className="button-content"
        >
          stats
        </div>
      </div>
    </div>
    <div
      className="HeroCard flex align-center"
    >
      <div
        className="invitePlayerButton"
      >
        <div
          className=""
        >
          <img
            alt="dva icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
        <div
          className="overlay"
        >
          <div
            className="plus-container flex align-center justify-center"
            onClick={[Function]}
          >
            <i
              className="fa fa-plus"
            />
          </div>
        </div>
      </div>
      <div
        className="content flex flex-column"
      >
        <div
          className="display-name"
        >
          You
        </div>
        <div
          className="sub-title"
        >
          <span>
            2400
             SR
          </span>
          <span>
             | 
          </span>
          <span>
            61.1
            % WR
          </span>
        </div>
      </div>
      <div
        className="button-primary"
        onClick={[Function]}
      >
        <div
          className="button-content"
        >
          stats
        </div>
      </div>
    </div>
    <div
      className="HeroCard flex align-center"
    >
      <div
        className="invitePlayerButton"
      >
        <div
          className=""
        >
          <img
            alt="zenyatta icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
        <div
          className="overlay"
        >
          <div
            className="plus-container flex align-center justify-center"
            onClick={[Function]}
          >
            <i
              className="fa fa-plus"
            />
          </div>
        </div>
      </div>
      <div
        className="content flex flex-column"
      >
        <div
          className="display-name"
        >
          You
        </div>
        <div
          className="sub-title"
        >
          <span>
            2400
             SR
          </span>
          <span>
             | 
          </span>
          <span>
            61.1
            % WR
          </span>
        </div>
      </div>
      <div
        className="button-primary"
        onClick={[Function]}
      >
        <div
          className="button-content"
        >
          stats
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeroRoles Component should match the snapshot no heroes 1`] = `
<div
  className="HeroRolesList card flex flex-column grow"
>
  <div
    className="header flex justify-between"
  >
    <h3>
      tank
    </h3>
    <img
      alt="tank icon"
      className="role-icon"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    className="body flex flex-column"
  >
    <div
      className="sub-title empty-list-message flex justify-center"
    >
      <div>
        No players online
      </div>
    </div>
  </div>
</div>
`;

exports[`HeroRoles Component should match the snapshot single hero 1`] = `
<div
  className="HeroRolesList card flex flex-column grow"
>
  <div
    className="header flex justify-between"
  >
    <h3>
      tank
    </h3>
    <img
      alt="tank icon"
      className="role-icon"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    className="body flex flex-column"
  >
    <div
      className="HeroCard flex align-center"
    >
      <div
        className="invitePlayerButton"
      >
        <div
          className=""
        >
          <img
            alt="soldier76 icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
        <div
          className="overlay"
        >
          <div
            className="plus-container flex align-center justify-center"
            onClick={[Function]}
          >
            <i
              className="fa fa-plus"
            />
          </div>
        </div>
      </div>
      <div
        className="content flex flex-column"
      >
        <div
          className="display-name"
        >
          You
        </div>
        <div
          className="sub-title"
        >
          <span>
            2400
             SR
          </span>
          <span>
             | 
          </span>
          <span>
            61.1
            % WR
          </span>
        </div>
      </div>
      <div
        className="button-primary"
        onClick={[Function]}
      >
        <div
          className="button-content"
        >
          stats
        </div>
      </div>
    </div>
  </div>
</div>
`;
